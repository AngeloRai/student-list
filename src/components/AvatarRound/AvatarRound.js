
const AvatarRound = ({ image, alt }) => {
  return (
    <img
      src={image}
      className="border-2 rounded-full w-24 m-2"
      alt={alt.img}
    />
  );
};

export default AvatarRound;
