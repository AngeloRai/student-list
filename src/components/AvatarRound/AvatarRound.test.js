import React from "react";
import { shallow, mount } from "enzyme";
import AvatarRound from "./AvatarRound";
import toJson from "enzyme-to-json";
// it("renders correctly", () => {
//   const wrapper = mount(<AvatarRound />);

// });

it("renders without crashing", () => {
  const wrapper = shallow(<AvatarRound image={image} alt={alt}/>);
  expect(toJson(wrapper)).toMatchSnapshot();
});

const image = "https://storage.googleapis.com/images/itiis.jpg";
const alt = { img: "student image" };

it("accepts user account props", () => {
  const wrapper = mount(<AvatarRound image={image} alt={alt} />);
  expect(wrapper.props().image).toEqual(image);
  expect(wrapper.props().alt).toEqual(alt);
});
it("should not render if no props received", () => {
  const wrapper = mount(<AvatarRound image={image} alt={alt} />);
  expect(wrapper.props().image).toEqual(image);
  expect(wrapper.props().alt).toEqual(alt);
});

it("renders an image", () => {
  const wrapper = shallow(<AvatarRound image={image} alt={alt}/>);
  expect(wrapper.find("img").prop("src")).toEqual(image);
});
