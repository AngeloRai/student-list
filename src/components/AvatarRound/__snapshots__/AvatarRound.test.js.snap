// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<img
  alt="student image"
  className="border-2 rounded-full w-24 m-2"
  src="https://storage.googleapis.com/images/itiis.jpg"
/>
`;
